contribute/design/1.x/csi/20190606-csi-volume-provisioning.md

```diff
@@ -0,0 +1,501 @@
+      * [Create a volume](#create-a-volume)
+      * [Delete the volume](#delete-the-volume)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Current Implementation -- Config](#current-implementation----config)
+      * [Current Implementation -- Volume Create](#current-implementation----volume-create)
+      * [Shortcomings with Current Implementation](#shortcomings-with-current-implementation)
+    * [Proposed Implementation](#proposed-implementation)
+      * [Volume Creation Workflow](#volume-creation-workflow)
+      * [Volume Deletion Workflow](#volume-deletion-workflow)
+    * [High Level Design](#high-level-design)
+      * [CStorVolumeConfigClass -- new custom resource](#cstorvolumeconfigclass----new-custom-resource)
+      * [CStorVolumeClaim -- new custom resource](#cstorvolumeclaim----new-custom-resource)
+      * [ConfigInjector -- new custom resource](#configinjector----new-custom-resource)
+      * [CStorVolume -- existing custom resource](#cstorvolume----existing-custom-resource)
+      * [CStorVolumeReplica -- existing custom resource](#cstorvolumereplica----existing-custom-resource)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This proposal charts out the design details to implement Container Storage 
+Interface commonly referred to as **CSI** to create and delete openebs volumes.
+
+## Motivation
+
+CSI implementation provides the necessary abstractions for any storage provider
+to provision volumes and related use-cases on any container orchestrator. This
+is important from the perspective of all OpenEBS storage engines, since it will 
+make all OpenEBS storage engines adhere to CSI specifications and hence abstract
+the same from inner workings of a container orchestrator.
+
+### Goals
+
+- Ability to create and delete a volume in a kubernetes cluster using CSI
+- Enable higher order applications to consume this volume
+
+### Non-Goals
+
+- Providing tunables w.r.t creating and deleting a volume
+- Coupling CSI implementation with any particular container orchestrator e.g. Kubernetes
+- Ability to resize a volume
+- Ability to take volume snapshots
+- Ability to clone a volume
+- Handling high availability in cases of node restarts
+- Handling volume placement related requirements
+
+## Proposal
+
+### User Stories
+
+#### Create a volume
+As an application developer I should be able to provide a volume that can be 
+consumed by my application. This volume should get created dynamically during 
+application creation time.
+
+#### Delete the volume
+As an application developer I should be able to delete the volume that was being
+consumed by my application. This volume should get deleted only when the 
+application is deleted.
```

Comment:
Should it be something like:
This volume can be deleted only after the application consuming it has been deleted.

Reply:
ok